Source/WebCore/html/parser/HTMLParserIdioms.h

```diff
@@ -117,7 +117,7 @@ template<typename CharacterType> inline bool isNotHTMLSpace(CharacterType charac
     return !isHTMLSpace(character);
 }
 
-inline bool isHTMLLineBreak(UChar character)
+template<typename CharacterType> inline bool isHTMLLineBreak(CharacterType character)
 {
     return character <= '\r' && (character == '\n' || character == '\r');
 }
```

Comment:
This doesn’t seem likely to be needed. Given the function is `inline`, I’d expect the same performance benefit without this change. Did you try the other change without this one?

Reply:
I haven't yet but I will.

Reply:
You're right, making the function templated is not needed for the perf win, only using the lambdas.

That said, if the function is not templated, it makes it really not obvious why we're using a lambda.

If you write this:
```
auto proposedValueWithoutLineBreaks = proposedValue.removeCharacters([](auto character) {
    return isHTMLLineBreak(character);
});
```

and `isHTMLLineBreak()` is templated, it makes it clearer we're using the lambda to call the right isHTMLLineBreak template specialization.

Otherwise, you'd have to write:
```
auto proposedValueWithoutLineBreaks = proposedValue.removeCharacters(isHTMLLineBreak<UChar>);
```

Which makes it clear you'll use the UChar specialization only.

If we don't make the function templated, it is a lot less clear why:
```
auto proposedValueWithoutLineBreaks = proposedValue.removeCharacters([](auto character) {
    return isHTMLLineBreak(character);
});
```

is faster than the simpler:
```
auto proposedValueWithoutLineBreaks = proposedValue.removeCharacters(isHTMLLineBreak);
```

Reply:
To me the issue is inlining, not overloading.

Reply:
removeCharacters can’t do inlining when passed a function name, right?